src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1610,6 +1610,16 @@ RangeTableListContainsOnlyReferenceTables(List *rangeTableList)
 				return false;
 			}
 		}
+		else if (rangeTableEntry->rtekind == RTE_FUNCTION)
+		{
+			List *functionList = rangeTableEntry->functions;
+
+			if (!(list_length(functionList) == 1 &&
+				  ContainsReadIntermediateResultFunction(linitial(functionList))))
+			{
+				return false;
+			}
+		}
```

Comment:
Would it make sense to add the other types of recurring tuples as well?

Reply:
I thought about that, but, I guess:

*  `RECURRING_TUPLES_EMPTY_JOIN_TREE` is not applicable here? Without `FROM` clause, I wasn't able to write a query with `GROUP BY`s.
* `RECURRING_TUPLES_FUNCTION`: I thought there is not much value on implementing that, but let me add that since people using Citus in various ways that I cannot think of 😄 

I'll add them now

Reply:
> is not applicable here? 

This walker function may not be the right thing to do then. You could easily have `(SELECT 1 AS x) ref` as one of the items in the join tree.